File: openmemory/integrations/kilocode/kilocode_memory_client.py
from typing import Any, Dict, Optional
import requests

class KiloCodeMemoryClient:
    def __init__(self, base_url: str, user_id: str, app: str, headers: Optional[Dict[str, str]] = None):
        self.base_url = base_url.rstrip('/')
        self.user_id = user_id
        self.app = app
        self.headers = headers or {"content-type": "application/json"}

    def add_memory(self, text: str, metadata: Optional[Dict[str, Any]] = None, infer: bool = True):
        payload = {
            "user_id": self.user_id,
            "text": text,
            "metadata": metadata or {},
            "infer": infer,
            "app": self.app,
        }
        res = requests.post(f"{self.base_url}/api/v1/memories", json=payload, headers=self.headers)
        res.raise_for_status()
        return res.json()

    def list_memories(self, search_query: Optional[str] = None, page: int = 1, size: int = 10,
                      sort_column: Optional[str] = None, sort_direction: Optional[str] = None):
        params = {"user_id": self.user_id, "page": page, "size": size}
        if search_query:
            params["search_query"] = search_query
        if sort_column:
            params["sort_column"] = sort_column
        if sort_direction:
            params["sort_direction"] = sort_direction
        res = requests.get(f"{self.base_url}/api/v1/memories", params=params, headers=self.headers)
        res.raise_for_status()
        return res.json()

File: openmemory/integrations/kilocode/kilocode-memory-client.ts
export type CreateMemoryRequest = {
  user_id: string;
  text: string;
  metadata?: Record<string, any>;
  infer?: boolean;
  app?: string; // project name
};

export class KiloCodeMemoryClient {
  constructor(
    private baseUrl: string,
    private userId: string,
    private app: string,
    private opts?: { headers?: Record<string, string> }
  ) {}

  async addMemory(text: string, metadata: Record<string, any> = {}, infer = true) {
    const body: CreateMemoryRequest = {
      user_id: this.userId,
      text,
      metadata,
      infer,
      app: this.app,
    };

    const res = await fetch(`${this.baseUrl}/api/v1/memories`, {
      method: 'POST',
      headers: {
        'content-type': 'application/json',
        ...(this.opts?.headers || {}),
      },
      body: JSON.stringify(body),
    });
    if (!res.ok) throw new Error(`addMemory failed: ${res.status}`);
    return res.json();
  }

  async listMemories(params: {
    search_query?: string;
    page?: number;
    size?: number;
    sort_column?: 'memory' | 'app_name' | 'created_at';
    sort_direction?: 'asc' | 'desc';
  } = {}) {
    const q = new URLSearchParams({ user_id: this.userId });
    if (params.search_query) q.set('search_query', params.search_query);
    if (params.page) q.set('page', String(params.page));
    if (params.size) q.set('size', String(params.size));
    if (params.sort_column) q.set('sort_column', params.sort_column);
    if (params.sort_direction) q.set('sort_direction', params.sort_direction);

    const res = await fetch(`${this.baseUrl}/api/v1/memories?${q.toString()}`);
    if (!res.ok) throw new Error(`listMemories failed: ${res.status}`);
    return res.json();
  }
}

